### ISSUE TITLE
Distinguish between physical and logical operations

### ISSUE THREAD

**Issue by danielkhan** [open-telemetry/opentelemetry-specification#110]
It'll be helpful to clarify whether this is a physical operation or logical one. My suggestion: only mark `SpanKind = Client` for the actual physical operation, for logical operation use `SpanKind = Internal`.

For example, the user would have `requests.get('https://www.wikipedia.org/wiki/Rabbit')`, which results in two physical HTTP requests - one 301 redirection, another 200 okay. The expected result is:
```
Span(SpanKind = Internal, http.status_code = 200, ...)
    Span(SpanKind = Client, http.status_code = 301, ...)
    Span(SpanKind = Client, http.status_code = 200, ...)
```

Please refer to https://github.com/census-instrumentation/opencensus-python/issues/627 for more background.

_Originally posted by @reyang in https://github.com/open-telemetry/opentelemetry-specification/pull/82_

**Comment by untitaker** [comment]
I am not sure if spankind is a useful discriminator. IMO it would make sense to have a freeform field to specify which exact instrumentation created this span (with the possible values "python-requests" and "python-httplib" in @danielkhan's example). This should clear the confusion and also serve as a string to search for if the user has to dive into code to understand why a span was created in the first place

**Comment by Oberon00** [comment]
> which exact instrumentation created this span

You will be pleased to find out that this is covered in https://github.com/open-telemetry/oteps/pull/16. 😃 Although I'm not sure if this can or should be used to solve that problem.

**Comment by jmacd** [comment]
Related: #51

**Comment by tedsuo** [comment]
Since we have merged in #263 I would like to close this for now, please reopen if you feel your concerns have not been addressed.

**Comment by untitaker** [comment]
Do I understand it correctly that http client instrumentation is supposed to create a single http-type span regardless of the amount of redirects?

If I really want to represent the raw network operations *with* parsed HTTP data, should I just make up my own component identifier and reuse the existing `http.*` tags?

**Comment by jmacd** [comment]
I don't think so @untitaker. I see this as a question about how to set `SpanKind` when redirects take place.

**Comment by untitaker** [comment]
right @jmacd I think we're roughly on the same page wrt what the question is. It's just unclear to me what the answer is.

**Comment by carlosalberto** [comment]
Bumping it to a future milestone.

**Comment by tigrannajaryan** [comment]
Do we still believe this is a must-have for GA? It is a guidance to instrumentation. If changed after GA it will not break any API or end-user code but may result in different trace composition. I think trace composition is not something we should consider part of our stability guarantees, hence I suggest we don't attempt to finalize such behavior before GA.
Given above reasoning I propose to remove ` release:required-for-ga` label.

**Comment by ocelotl** [comment]
I think this issue can be closed, the semantic conventions already specify how this scenario should be:

https://github.com/open-telemetry/semantic-conventions/blob/main/docs/http/http-spans.md#http-request-retries-and-redirects
